src/metabase/events/recent_views.clj

```diff
@@ -42,41 +36,40 @@
      :user-id user-id}
     (try
       (when object
-        (let [model    (audit-log/model-name object)
-              model-id (u/id object)
+        (let [model-id (u/id object)
               user-id  (or user-id api/*current-user-id*)]
-          (recent-views/update-users-recent-views! user-id model model-id)))
+          (recent-views/update-users-recent-views! user-id :model/Table model-id :view)))
       (catch Throwable e
         (log/warnf e "Failed to process recent_views event: %s" topic)))))
 
 (derive ::card-query-event :metabase/event)
 (derive :event/card-query ::card-query-event)
 
-
 (m/defmethod events/publish-event! ::card-query-event
   "Handle processing for a single card query event."
   [topic {:keys [card-id user-id context] :as _event}]
   (try
     (let [user-id  (or user-id api/*current-user-id*)]
       ;; we don't want to count pinned card views
       (when-not (#{:collection :dashboard} context)
-        (recent-views/update-users-recent-views! user-id "card" card-id)))
+        (recent-views/update-users-recent-views! user-id :model/Card card-id :view)))
     (catch Throwable e
       (log/warnf e "Failed to process recent_views event: %s" topic))))
 
 (derive ::legacy-card-event :metabase/event)
-(derive :event/card-create ::legacy-card-event)
+;; in practice, updating or creating a card will immediately trigger a card-read
 (derive :event/card-read ::legacy-card-event)
-(derive :event/card-update ::legacy-card-event)
 
 (m/defmethod events/publish-event! ::legacy-card-event
-  "Handle recent view processing for CRU (not D) events"
-  [topic {:keys [object user-id]}]
-  (try
-    (recent-views/update-users-recent-views! (or user-id api/*current-user-id*) :model/Card (:id object))
-    (catch Throwable e
-      (log/warnf e "Failed to process recent_views event: %s" topic))))
-
+  "Handle recent-view processing for card reads"
+  [topic {:keys [object-id user-id context]}]
+  (when (= context :question)
+    ;; GET /api/card/:id will trigger a card-read event. We don't want to count _pinned collection views_ as recent-views,
+    ;; so a collection with pinned questions will send a request with context==:collection, which we skip below:
```

Comment:
As of https://github.com/metabase/metabase/pull/43102 `:context` can be `:dashboard` too.
```
grep -r --include=\*.clj "event/card-read" src 

src/metabase/api/embed.clj:      (events/publish-event! :event/card-read {:object-id (:id <>), :user-id api/*current-user-id*, :context :question}))))
src/metabase/api/embed.clj:    (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard})))
src/metabase/api/embed.clj:    (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard})))
src/metabase/api/card.clj:        (events/publish-event! :event/card-read {:object-id (:id <>) :user-id api/*current-user-id* :context :question})))))
src/metabase/api/dashboard.clj:    (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard})))
src/metabase/api/dashboard.clj:    (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard})))
src/metabase/api/public.clj:    (events/publish-event! :event/card-read {:object-id (:id <>), :user-id api/*current-user-id*, :context :question})))
src/metabase/api/public.clj:      (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard}))))
src/metabase/api/public.clj:      (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard}))))
```

Should this comment also mention that card view with `context=dashboard` are also omitted from recent views? It implies that the only  purpose of this when clause is to exclude views where `context=collection`.

Reply:
Yeah I think updating comments so they're more informative totally worth it. I'll make the change.